src/analyzer.ts

```diff
@@ -154,6 +163,9 @@ export class Analyzer {
   private async _analyzeDocument(
       document: ParsedDocument<any, any>, maybeLocationOffset?: LocationOffset,
       maybeAttachedComment?: string): Promise<ScannedDocument> {
+    if (document == null) {
```

Comment:
We could do it upstream, but we need to return `null` somewhere around here in the case where there's a parse error. This is handled better in a downstream commit where we actually handle it before calling _analyzeDocument but it was expedient to do it here just so that parsing all polymer elements works at all.